src/derivative_operators/convolutions.jl

```diff
@@ -12,14 +12,23 @@ function convolve_interior!(x_temp::AbstractVector{T}, x::AbstractVector{T}, A::
     @assert length(x_temp)+2 == length(x)
     stencil = A.stencil_coefs
     coeff   = A.coefficients
-    mid = div(A.stencil_length,2)
+    # Upwind operators have a non-centred stencil
+
+    if use_winding(A)
+        mid = 1 + A.stencil_length%2
+    else
+        mid = div(A.stencil_length,2)
+    end
+
     for i in (1+A.boundary_point_count) : (length(x_temp)-A.boundary_point_count)
         xtempi = zero(T)
         cur_stencil = eltype(stencil) <: AbstractVector ? stencil[i-A.boundary_point_count] : stencil
         cur_coeff   = typeof(coeff)   <: AbstractVector ? coeff[i] : coeff isa Number ? coeff : true
         cur_stencil = use_winding(A) && cur_coeff < 0 ? reverse(cur_stencil) : cur_stencil
         for idx in 1:A.stencil_length
-            xtempi += cur_coeff * cur_stencil[idx] * x[i - mid + idx]
+            x_idx = use_winding(A) && cur_coeff < 0 ? x[i + mid - idx] : x[i - mid + idx]
+            xtempi += cur_coeff * cur_stencil[idx] * x_idx
+            # @show i, idx, cur_stencil[idx], i-mid+idx, x[i-mid+idx]
```

Comment:
delete

Reply:
?

Reply:
this comment. I assume you meant to put this as no longer WIP?